Fixes #82

1-alex98 opened the pull request.

1-alex98 committed:
Use long term credentials for custom ICE servers

Fixes #82

Geosearchef commented on faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java:
```diff
@@ -0,0 +1,80 @@
+@Service
+@Slf4j
+@Conditional(StandardIceServerProviderCondition.class)
+public class StandardIceServerProvider implements IceServersProvider {
```
What about LongTormCrendential(IceServer)Provider?

1-alex98 replied:
What is that????

Geosearchef replied:
A better name than standard ice server provider ;)

Geosearchef commented on faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java:
```diff
@@ -0,0 +1,80 @@
+
+    String userAuth = String.format("%s:%s", Instant.now().plusSeconds(ttl).getEpochSecond(), player.getLogin());
+
+    return new IceServerList(ttl, Instant.now(),
```
this and the userAuth/name should probably have the same time if someone tries to calculate the date of expiry using this and the ttl.

Geosearchef commented on faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java:
```diff
@@ -0,0 +1,80 @@
+
+    } catch (NoSuchAlgorithmException | InvalidKeyException e) {
+      log.error("Could not build secret key", e);
+      throw new RuntimeException(e);
```
Is this copied from my test code? I don't think we should just throw a runtime exception here, instead just don't return an ICE server for this url and log an error.

1-alex98 replied:
yes it is and you even did System.exit(-1) but i thought that is a little over dramatic

Geosearchef commented:
Thx for implementing this!

axel1200 commented:
I made you a new image with the bug fixed "ice-test"

micheljung committed:
Let tests download GeoIp database from our own servers

It's probably being delivered by the CDN Cloudflare and it happened in
the past that the Travis servers got blocked.

axel1200 commented:
Anymore discusiions on this?

micheljung commented:
https://github.com/travis-ci/travis-ci/issues/10249
